### DISCUSSION TITLE
Need advice on if/how MIDI can be routed from a Wine program to the host

### DISCUSSION THREAD

**Discussion by dhasial** [DISTRHO/Cardinal#498] [Q&A]
My host's audio engine is Pipewire.

I'm running Cardinal on the host and Ableton under wine, and I wanted to see if the repo owners or anyone else had an idea of how I would do this. I've tried using qpwgraph to connect Ableton's midi outputs to Cardinal but the only outputs Ableton is showing on it are for audio. I've also gone into the Ableton setting to try enabling midi sends for the midi tracks, then connecting those to Cardinal's cv_in and events_in, but no luck in getting it to work yet.

**Comment by dromer** [comment]
One problem I have seen with wine is that once you have opened a program it never updates the midi ports. You'd have to first start the Linux program and after start your program in wine.

However, in your case wouldn't it make more sense to install the Windows version of Cardinal and just run the VST2/3 inside of Ableton?
